leader-election-impl/src/main/java/com/palantir/leader/PaxosLeadershipEventRecorder.java

```diff
@@ -27,6 +30,7 @@
 
     private final String leaderId;
     private final LeadershipEvents events;
+    private final List<LeadershipObserver> leadershipObservers;
```

Comment:
Seems a little odd that this has a `List<LeadershipObserver>` and the only implementation of that contains a `List<Runnable>`.  I'd try to only have one level of multiplicity.

Reply:
Makes sense to me; I changed it to keep an Optional<LeadershipObserver> rather than keeping a list of observers.